src/OpenTelemetry.OpAmp.Client/Internal/FrameProcessor.cs

```diff
@@ -13,10 +13,10 @@ namespace OpenTelemetry.OpAmp.Client.Internal;
 
 internal sealed class FrameProcessor
 {
-    private readonly ConcurrentDictionary<Type, IReadOnlyList<IOpAmpListener>> listeners = [];
+    private readonly ConcurrentDictionary<Type, IReadOnlyList<object>> listeners = [];
```

Comment:
I think this is Advent-of-Code recency making me notice this, but given that we only iterate over the list or change the size +/- 1 with object identity, I wonder if `I{ReadOny}Set<T>` backed with a `HashSet<T>` would be a bit more efficient?

Then could be something like the below?

```csharp
(_, set) => [.. set, listener];
(_, set) => [.. set.Except(listener)];
```

Reply:
I'm not adverse to looking at that, although perhaps we could make that a follow up PR after reviewing benchmarks etc. I was also thinking that perhaps a linked list approach might also work, similar to the internals of ActivityListeners I believe.